server/segment_rpc.go

```diff
@@ -434,16 +435,24 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64,
 		return nil, err
 	}
 
+	// timeout for the whole HTTP call: segment upload, transcoding, reading response
+	httpTimeout := common.HTTPTimeout
 	// set a minimum timeout to accommodate transport / processing overhead
-	dur := common.HTTPTimeout
 	paddedDur := 4.0 * seg.Duration // use a multiplier of 4 for now
-	if paddedDur > dur.Seconds() {
-		dur = time.Duration(paddedDur * float64(time.Second))
+	if paddedDur > httpTimeout.Seconds() {
+		httpTimeout = time.Duration(paddedDur * float64(time.Second))
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), dur)
+	// timeout fot the segment upload, until HTTP returns OK 200
+	uploadTimeout := time.Duration(0.5 * seg.Duration * float64(time.Second)) // use 0.5 multiplier
+	if uploadTimeout <= 0 {
```

Comment:
Ah I see... 

**nit:** Another possibility is to not have any upload timeout logic when the segment duration is `<= 0` then. e.g. start `sendReqWithTimeout` with something like `if timeout <= 0 { return httpClient.Do(req) }`

But I think we might always want to have a default upload timeout here in case the segment duration is undefined. WDYT of defining a `common.SegmentUploadTimeout` just like the `common.HTTPTimeout` above, which we use by default in case the segment duration is `<= 0` (instead of `== httpTimeout` which is a noop)

Reply:
Good idea, added `common.SegmentUploadTimeout` and set it to `2s`